plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -410,6 +411,35 @@ private static <K, V> V getWithValueHolder(Cache<K, AtomicReference<V>> cache, K
         return value;
     }
 
+    private static <K, V> V getWithValueHolder(
```

Comment:
the `getWithValueHolder` (overload) has the name it has because we couldn't find a good name for that logic

i think the key aspect of this new method is that it loads or loads incrementally, so we should be able to give it a nice name. `getIncrementally`?